tools/pythonpkg/duckdb/experimental/spark/sql/dataframe.py

```diff
@@ -903,9 +904,14 @@ def dropDuplicates(self, subset: Optional[List[str]] = None) -> "DataFrame":
         +-----+---+------+
         """
         if subset:
-            raise ContributionsAcceptedError
+            rn_col = f"tmp_col_{uuid.uuid1().hex}"
+            window_spec = f"OVER(PARTITION BY {', '.join(subset)} ORDER BY {', '.join(subset)}) AS {rn_col}"
```

Comment:
Can we add a test with names that contain keywords?
I suspect these need to be quoted here